plugins/cpp/parser/src/clangastvisitor.h

```diff
@@ -141,6 +141,50 @@ class ClangASTVisitor : public clang::RecursiveASTVisitor<ClangASTVisitor>
     return b;
   }
 
+
+  bool	TraverseCXXForRangeStmt (clang::CXXForRangeStmt *forRangeStmt){
+    return Base::TraverseCXXForRangeStmt(forRangeStmt);
+  }
+
+  bool VisitCXXForRangeStmt(clang::CXXForRangeStmt *forRangeStmt)
+  {
+    model::CppAstNodePtr astNode = std::make_shared<model::CppAstNode>();
+    
+    clang::DeclRefExpr* declRefExpr = llvm::dyn_cast<clang::DeclRefExpr>(forRangeStmt->getRangeInit());
+
+    clang::Stmt* stmt = llvm::dyn_cast<clang::Stmt>(forRangeStmt->getRangeInit());
+
+    clang::CallExpr* callExpr = llvm::dyn_cast<clang::CallExpr>(stmt);
+
+    if(declRefExpr){ //lvalue Var
+      VisitDeclRefExpr(declRefExpr);
+    }
+
+    if(callExpr){ //lvalue Function
+      VisitCallExpr(callExpr);
+    }
+    
+    astNode->astValue =  getSourceText(
+      _clangSrcMgr,
+      forRangeStmt->getRangeStmt()->getBeginLoc(),
+      forRangeStmt->getRangeStmt()->getEndLoc(),
+      true);    
+
+    astNode->location = getFileLoc(forRangeStmt->getBeginLoc(), forRangeStmt->getEndLoc());
+
+    const clang::VarDecl* vd = llvm::dyn_cast<clang::VarDecl>(forRangeStmt->getRangeStmt()->getSingleDecl());
+    astNode->entityHash = util::fnvHash(getUSR(vd));
+
+    astNode->symbolType = model::CppAstNode::SymbolType::Variable;
+
+    astNode->astType = model::CppAstNode::AstType::Read;
+
+    astNode->id = model::createIdentifier(*astNode);
+
+    _astNodes.push_back(astNode);
```

Comment:
Since you also call `VisitDeclRefExpr`, this seems to be redundant.

Reply:
Removed.